aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2141,36 +2141,56 @@ std::vector<Tensor> meshgrid(TensorList tensors,
   int64_t size = tensors.size();
   TORCH_CHECK(size > 0, "meshgrid expects a non-empty TensorList");
 
+  std::vector<std::reference_wrapper<const Tensor>> tensor_refs(tensors.begin(),
+                                                                tensors.end());
+  bool swap_first_and_second_tensors;
+
   if (!indexing.has_value()) {
     TORCH_WARN_ONCE("In an upcoming release, it will be required to pass the "
                     "indexing argument to torch.meshgrid.");
     indexing = "ij";
+    swap_first_and_second_tensors = false;
   } else {
-    TORCH_CHECK(*indexing == "ij", "Unsupported indexing: ", *indexing);
+    if (*indexing == "xy") {
+      // We can only swap if there are multiple tensors.
+      swap_first_and_second_tensors = size >= 2;
+      if (swap_first_and_second_tensors) {
+        std::swap(tensor_refs[0], tensor_refs[1]);
+      }
+    } else {
+      TORCH_CHECK(*indexing == "ij", "Unsupported indexing: ", *indexing);
```

Comment:
Comment here explaining why it's OK to just check 'ij' (because 'xy' is caught above)

Reply:
done